new-components/scanners/nancy/internal/transformer/transformer.go

```diff
@@ -0,0 +1,455 @@
+package transformer
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/go-errors/errors"
+	"github.com/jonboulle/clockwork"
+	"github.com/package-url/packageurl-go"
+	"github.com/smithy-security/pkg/env"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"github.com/smithy-security/smithy/sdk/component"
+	ocsffindinginfo "github.com/smithy-security/smithy/sdk/gen/ocsf_ext/finding_info/v1"
+	ocsf "github.com/smithy-security/smithy/sdk/gen/ocsf_schema/v1"
+)
+
+type (
+
+	// NancyTransformerOption allows customising the transformer.
+	NancyTransformerOption func(b *NancyTransformer) error
+
+	// NancyOut represents the output of a nancy run that we care about.
+	NancyOut struct {
+		Vulnerable    []NancyAdvisory `json:"vulnerable"`
+		Audited       []interface{}
```

Comment:
Why are these so weakly typed?

Reply:
they are unused at the moment, changed to any and added a comment